compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract/Templates.hs

```diff
@@ -88,6 +89,61 @@ getTemplateDocs DocCtx{..} typeMap templateInstanceMap =
                       [] -> [] -- catching the dummy case here, see above
                       _other -> error "getFields: found multiple constructors"
 
+-- | Build interface docs up from class docs.
+getInterfaceDocs :: DocCtx
+    -> MS.Map Typename ADTDoc -- ^ maps template names to their ADT docs
+    -> [InterfaceDoc]
+getInterfaceDocs DocCtx{..} typeMap =
+    map mkInterfaceDoc $ Set.toList dc_interfaces
+  where
+    -- The following functions use the type map and choice map in scope, so
+    -- defined internally, and not expected to fail on consistent arguments.
+    mkInterfaceDoc :: Typename -> InterfaceDoc
+    mkInterfaceDoc name = InterfaceDoc
+      { if_anchor = ad_anchor tmplADT
+      , if_name = ad_name tmplADT
+      , if_descr = ad_descr tmplADT
+      , if_choices = map mkChoiceDoc choices
+      , if_methods = [] -- TODO (drsk)
+      }
+      where
+        tmplADT = asADT name
+        choices = Set.toList . fromMaybe Set.empty $ MS.lookup name dc_choices
+
+    mkChoiceDoc :: Typename -> ChoiceDoc
+    mkChoiceDoc name = ChoiceDoc
+      { cd_name = ad_name choiceADT
+      , cd_descr = ad_descr choiceADT
+      -- assumes exactly one constructor (syntactic in the template syntax), or
+      -- uses a dummy value otherwise.
+      , cd_fields = getFields choiceADT
+      }
+      where choiceADT = asADT name
+
+    asADT n = fromMaybe dummyDT $
+              MS.lookup n typeMap
+    -- returns a dummy ADT if the choice argument is not in the local type map
+    -- (possible if choice instances are defined directly outside the template).
+    -- This wouldn't be necessary if we used the type-checked AST.
```

Comment:
side note: I think we do use the typechecked AST these days so this comment (copied from how we do it for templates) might also be out of date.

Reply:
Dropped the comment. I did leave the default though, thinking that throwing an exception and aborting if the type is not found is probably more confusing than anything for the user.